components/brave_shields/browser/custom_subscription_download_manager.cc

```diff
@@ -0,0 +1,213 @@
+#include "content/public/browser/browser_thread.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "storage/browser/blob/blob_data_handle.h"
+#include "storage/browser/blob/write_blob_to_file.h"
+
+namespace brave_shields {
+
+namespace {
+
+const net::NetworkTrafficAnnotationTag
+    kBraveShieldsCustomSubscriptionTrafficAnnotation =
+        net::DefineNetworkTrafficAnnotation("brave_shields_custom_subscription",
+                                            R"(
+        semantics {
+          sender: "Brave Shields"
+          description:
+            "Brave periodically downloads updates to third-party filter lists "
+            "added by users on brave://adblock."
+          trigger:
+            "After being registered in brave://adblock, any enabled filter "
+            "list subscriptions will be updated in accordance with their "
+            "`Expires` field if present, or daily otherwise. A manual refresh "
+            "for a particular list can also be triggered in brave://adblock."
+          data: "The URL endpoint provided by the user in brave://adblock to "
+            "fetch list updates from. No user information is sent."
+          destination: BRAVE_OWNED_SERVICE
+        }
+        policy {
+          cookies_allowed: NO
+          setting:
+            "This request cannot be disabled in settings. However it will "
+            "never be made if the corresponding entry is removed from the "
+            "brave://adblock page's custom list subscription section."
+          policy_exception_justification: "Not yet implemented."
+        })");
+
+}  // namespace
+
+CustomSubscriptionDownloadManager::CustomSubscriptionDownloadManager(
+    download::DownloadService* download_service,
+    AdBlockSubscriptionServiceManager* subscription_manager,
+    scoped_refptr<base::SequencedTaskRunner> background_task_runner)
+    : download_service_(download_service),
+      is_available_for_downloads_(true),
+      background_task_runner_(background_task_runner),
+      subscription_manager_(subscription_manager) {}
+
+CustomSubscriptionDownloadManager::~CustomSubscriptionDownloadManager() =
+    default;
+
+void CustomSubscriptionDownloadManager::StartDownload(const GURL& download_url,
+                                                      bool from_ui) {
+  download::DownloadParams download_params;
+  download_params.client = download::DownloadClient::CUSTOM_LIST_SUBSCRIPTIONS;
+  download_params.guid = base::GenerateGUID();
+  download_params.callback =
+      base::BindRepeating(&CustomSubscriptionDownloadManager::OnDownloadStarted,
+                          ui_weak_ptr_factory_.GetWeakPtr(), download_url);
+  download_params.traffic_annotation = net::MutableNetworkTrafficAnnotationTag(
+      kBraveShieldsCustomSubscriptionTrafficAnnotation);
+  download_params.request_params.url = download_url;
+  download_params.request_params.method = "GET";
+  if (from_ui) {
```

Comment:
from_ui seems always true?

Reply:
This is a good catch. I hadn't exposed it at a high enough level, I added it to the signature of `RefreshSubscription` as well and now it's false in the bound callback for background refreshes.